### ISSUE TITLE
ShuffleNet_ExLarge中SE_block问题

### ISSUE THREAD

**Issue by Hucley** [megvii-model/ShuffleNet-Series#33]
在116-119行
‘’’
        if not self.has_proj:
            x = self.relu(x_proj + x)
        x = torch.cat((proj, x), dim=1)
‘’’
如果使用 self.has_proj==True，此时stride=2,所以就不使用residual形式，直接concat  proj和seblock的输出x,相反则使用residual形式，我这边在自己的数据集使用（数据集无问题，其他网络收敛正常）这种方法后一直未收敛；但当在stride=2时，改为不经过se_block直接 concat proj和x则能正常收敛（即stride=2时不使用SE_Block也不使用reisual形式，stride=1不变（使用SENet_Block和residual形式））。
但在你们的shufflenetv2+中使用SE_Block后则一直保持非residual形式，我这边直接初始化训练也正常；确认了下官方SENet代码中当strde=2或者前后channel不相等时，会使用相应的卷积进行proj使特征图大小和channe保持一致后在形成residual形式，使其整个网络在使用SE_Block后都保持residual形式；
所以我直接用你们的公开的代码初始化训练一直不收敛，是不是因为SE_Block前后使用形式不一致（resiual和非residual形式混合使用）导致网络不稳定，从而影响收敛性。不知你们在训练该网络时有没有遇到这种情况，希望能获得意见。

**Comment by nmaac** [comment]
Hi, @Hucley ,
我们除了在pytorch还在其他framework上train过，都是可以正常收敛的，没有遇到过不收敛的情况。

**Comment by Hucley** [comment]
ok, thx
